esp/scm/ws_resources.ecm

```diff
@@ -25,11 +25,35 @@ ESPStruct HPCCService
     bool TLSSecure;
 };
 
+ESPStruct ServicePorts
+{
+    string Name;
+    string Protocol;
+    unsigned Port;
+    unsigned NodePort;
+    unsigned TargetPort;
+};
+
+//The ExternalIPsEx or ClusterIPsEx is used when ExternalIPs or ClusterIPs is empty.
+//The value is based on the existing https://github.com/kubernetes/kubernetes.
+ESPStruct ServiceConnection
+{
+    string Type;
+    ESParray<string> ExternalIPs;
+    string ExternalIPsEx; //Use this field when ExternalIPs is empty. Set to '<none>' for spec type 'ClusterIP' and spec type 'NodePort';
+                          //set to '<pending>' for spec type 'LoadBalancer'; leave it empty for spec type 'ExternalName';
+                          //set to '<unknown>' for unknown spec types.
+    ESParray<string> ClusterIPs;
+    string ClusterIPsEx; //when ClusterIPs is empty, set to '<none>'.
```

Comment:
The Ex on the name is a little confusing.  I tend to think of Ex as an extended version of something.
If the only purpose is to say "<none>" when the IP list is empty, is it really needed?

Reply:
I may remove it. We may add a field if the value could not be '<none>'.